WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -935,21 +935,30 @@ - (NSPredicate *)fetchRequestPredicateForHideableBlogs
      -[Blog supports:BlogFeatureVisibility] should match this, but the logic needs
      to be duplicated because core data can't take block predicates.
      */
-    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
-    AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
-    WPAccount *defaultAccount = [accountService defaultWordPressComAccount];
-
+    WPAccount *defaultAccount = [self defaultWordPressComAccount];
     return [NSPredicate predicateWithFormat:@"account != NULL AND account = %@", defaultAccount];
 }
 
 - (NSPredicate *)fetchRequestPredicateForVisibleBlogs
 {
-    return [NSPredicate predicateWithFormat:@"visible = YES"];
+    WPAccount *defaultAccount = [self defaultWordPressComAccount];
+    if (defaultAccount) {
+        return [NSPredicate predicateWithFormat:@"visible = YES AND account = %@", defaultAccount];
```

Comment:
changed to 
`return [NSPredicate predicateWithFormat:@"visible = YES AND (account = %@ OR account = NULL)", defaultAccount];` 
 tested it with a wp.com account and a self-hosted site logged in and it shows up.